### PR TITLE
Use canary quay images to test latest sidecars

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
Use the public canary quay images until we redo our CI pipeline for latest testing

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml [hunk 1]
```diff
@@ -7,14 +7,14 @@ images:
   newName: gcr.io/gke-release-staging/gcp-compute-persistent-disk-csi-driver
   newTag: "latest"
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: hmmm this is a problem too isn't it?

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: maybe we need to enable GCE_PD_DO_DRIVER_BUILD?

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: yeah this tag is really old now - from sep 9.

Enabling the build flag isn't so easy - then we also need to add logic to go and parse this kustomization file and swap out "latest" for whatever the new build tag is... Though that logic already exists for the pre-submits so maybe its not so hard to port over to this file too. Want to do that or follow up later with it?

We should really figure out what to do about the test tag versions

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: From a quick glance, I think our k8s-integration.go should be able to handle substituting the image properly when we set the flag. It takes [both overlay and registry](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/driver.go#L15) as an argument. So no extra changes needed here.

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: ack. are you adding that change to this PR?

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/kustomization.yaml]: no, setting DO_DRIVER_BUILD is not in this repo. I'll send out a separate pr for that

**Pull request merged**